### PR TITLE
Add basic Archlinux support

### PR DESCRIPTION
**Pull request by MerlinDMC**
This adds `install.sh` and `joyent_rc.local` support for Archlinux.
This also adds a systemd unit file for the `joyent_rc.local` script.

### CODE REVIEW

File: src/lib/smartdc/archlinux [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2015, Joyent, Inc. All rights reserved.
+#
+# Archlinux specific setup
+
+# load common functions and vars
+. ./lib/smartdc/common.lib
```
Comment (chorrell) [inline src/lib/smartdc/archlinux]: Can you change this to be absolute? e.g.:

`. /lib/smartdc/common.lib`

Comment (MerlinDMC) [inline src/lib/smartdc/archlinux]: That was a whoopsie.

changed and squashed.

**Comment by MerlinDMC** [comment]
Do you need anything in addition to the changes I squashed into the last commit?

**Comment by chorrell** [comment]
Hi.

No it looks good, I just haven't had the time to run this through my testing yet. I plan on doing that, and merging, next week.

**Comment by chorrell** [comment]
LGTM

**Pull request merged**